src/script/InitDelegateeDeposits.s.sol

```diff
@@ -0,0 +1,152 @@
+    string memory _path = string.concat(_root, "/src/script/addresses.json");
+    string memory _json = vm.readFile(_path);
+
+    address[] memory _delegateeAddresses = stdJson.readAddressArray(_json, ".addresses");
+    return _delegateeAddresses;
+  }
+
+  /// @notice Retrieves deposit IDs for a list of delegatee addresses.
+  /// @param _delegateeAddresses Array of delegatee addresses to check.
+  /// @return An array of deposit IDs corresponding to each delegatee address.
+  function getDepositIdsForDelegateeAddresses(address[] memory _delegateeAddresses)
+    public
+    virtual
+    returns (uint256[] memory)
+  {
+    uint256 _delegateeAddressesLength = _delegateeAddresses.length;
+    uint256[] memory _depositIds = new uint256[](_delegateeAddressesLength);
+
+    bytes[] memory _data = new bytes[](_delegateeAddressesLength);
+    for (uint256 i = 0; i < _delegateeAddressesLength; i++) {
+      _data[i] = abi.encodeWithSelector(IGovLst.depositForDelegatee.selector, _delegateeAddresses[i]);
+    }
+
+    bytes[] memory _results = govLst.multicall(_data);
+    for (uint256 i = 0; i < _delegateeAddressesLength; i++) {
+      _depositIds[i] = abi.decode(_results[i], (uint256));
+    }
+    return _depositIds;
+  }
+
+  /// @notice Filters delegatee addresses to identify those needing deposit initialization.
+  /// @param _delegateeAddresses Array of all delegatee addresses.
+  /// @param _depositIds Array of deposit IDs corresponding to each delegatee address.
+  /// @return _addressesToInit An array of addresses that need initialization.
+  /// @return _numOfAddressesToInit The count of addresses that need initialization.
+  function filterDelegateeAddresses(address[] memory _delegateeAddresses, uint256[] memory _depositIds)
+    public
+    view
+    virtual
+    returns (address[] memory, uint256)
+  {
+    address[] memory _addressesToInit = new address[](_delegateeAddresses.length);
+    uint256 _numOfAddressesToInit = 0;
+
+    for (uint256 i = 0; i < _delegateeAddresses.length; i++) {
+      if (_depositIds[i] == 0 || _depositIds[i] == DEFAULT_DEPOSIT_ID) {
+        _addressesToInit[_numOfAddressesToInit] = _delegateeAddresses[i];
+        _numOfAddressesToInit++;
+      }
+    }
+    return (_addressesToInit, _numOfAddressesToInit);
+  }
+
+  /// @notice Initializes deposits for delegatee addresses in batches using multicall.
+  /// @param _addressesToInit Array of delegatee addresses that need initialization.
+  /// @param _numOfAddressesToInit Number of addresses to initialize.
+  /// @return _allResults An array of results from the initialization calls.
+  /// @return _batchCount The number of batches processed.
+  function callFetchOrInitializeDepositForDelegatee(address[] memory _addressesToInit, uint256 _numOfAddressesToInit)
+    public
+    virtual
+    returns (bytes[] memory, uint256)
+  {
+    uint256 _totalAddresses = _numOfAddressesToInit;
+    uint256 _batchCount = (_totalAddresses + BATCH_SIZE - 1) / BATCH_SIZE;
```

Comment:
This line is not what I would have expected. Does it definitely do what we want? I would have expected something like:

```solidity
uint256 _batchCount = _totalAddresses / BATCH_SIZE;
if (_totalAddresses % BATCH_SIZE > 0) {
    _batchCount++;
}
```

Does this line produce the same results?

Reply:
yes, that results in the same thing and is much easier to undertsand. Updated in https://github.com/withtally/stGOV/pull/91/commits/a174170819e02edf43a2baa11d7887d23bc27c9d